pkg/provisioner/ironic/ironic.go

```diff
@@ -1613,6 +1617,30 @@ func (p *ironicProvisioner) Delete() (result provisioner.Result, err error) {
 		return p.setMaintenanceFlag(ironicNode, true)
 	}
 
+	if ironicNode.PowerState == powerOn {
+		_, err := p.hardPowerOff()
+		if err != nil {
+			result.Dirty = true
+			switch err.(type) {
+			case HostLockedError:
+				p.log.Info("could not power off host, busy")
+				return operationContinuing(powerRequeueDelay)
```

Comment:
On a second thought, it's probably better to use `retryAfterDelay` here, since it's currently used when a retry is required after an error. `operationContinuing` has technically the same behavior, but it is currently adopted in normal scenarios where an error is not triggered (ie waiting for some event)

Reply:
I agree. So, going back to retryAfterDelay.